platform/pulpcore/app/viewsets/task.py

```diff
@@ -19,18 +20,19 @@ class Meta:
         fields = ['state', 'worker__name', 'tags']
 
 
-class TaskViewSet(NamedModelViewSet):
+class TaskViewSet(mixins.RetrieveModelMixin,
+                  mixins.ListModelMixin,
+                  GenericNamedModelViewSet):
```

Comment:
Do you have any objections to create a base class `ReadNamedModelViewSet` like [this one](https://github.com/asmacdo/pulp/blob/3104d240fc7d4e2a6a882420b095a43edda09686/platform/pulpcore/app/viewsets/base.py#L149)? I don't think we would need all the combinations of different mixins, likely there will be several common combinations. What do you think?

Reply:
I think that those base classes only make sense in certain circumstances. For instance, Repositories, Importers, Publishers, Distributions, (and Publications?) will all share a set of behaviors. Synchronous create, list, and retrieve. Async update and delete. To me, it makes sense to have a base viewset for those because it is applicable to so many viewsets. I don't think we have that here, so I'd rather use mixins directly.

Mostly, I think that direct mixin use is more clear, less error prone, and simpler to user. You don't have to hunt for the correct base class, you don't have to guess what class names mean, and you don't have to leave viewsets.base.py open to understand the behaviors of your viewset.

Reply:
I'm not convinced but not against either :)